tests/integration/plugins/test_plugins.py

```diff
@@ -236,3 +240,74 @@ async def test_knn_training_search(ops_test: OpsTest) -> None:
         except RetryError:
             # The search should fail if knn_enabled is false
             assert not knn_enabled
+
+
+@pytest.mark.abort_on_fail
+async def test_mlcommons_upload_and_predict_model(ops_test: OpsTest) -> None:
+    """Uploads and runs the model."""
+
+    async def __wait_model_task(task_id: str):
+        for _ in range(5):
+            model_id = await mlcommons_wait_task_model(ops_test, app, leader_unit_ip, task_id)
+        assert model_id is not None
+        return model_id
+
+    app = (await app_name(ops_test)) or APP_NAME
+
+    leader_unit_ip = await get_leader_unit_ip(ops_test, app=app)
+
+    task_id = (
+        await mlcommons_upload_model(
+            ops_test,
+            app,
+            leader_unit_ip,
+            model_config={
+                "name": "huggingface/sentence-transformers/all-MiniLM-L12-v2",
+                "version": "1.0.1",
+                "model_format": "TORCH_SCRIPT",
+            },
+        )
+    ).get("task_id", None)
+
+    assert task_id is not None
```

Comment:
Could you add an explanatory message in all your asserts? (including the ones after this block on code)
```suggestion
    assert task_id is not None, "The task_id is None"
```